source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,155 @@
+=========================
+Hooks Extension Framework
+=========================
+
+What is the Hooks Extension Framework?
+=======================================
+
+Based on the `open-closed principle`_, this framework aims to extend the Open edX platform in a maintainable way without modifying its core. The main goal is to leverage the existing extension capabilities provided by the plugin architecture, allowing developers to implement new features to fit customer needs while reducing the need for core modifications and minimizing maintenance efforts.
+
+Hooks are a list of places in the Open edX platform where externally defined functions can take place. These functions may alter what the user sees or experiences on the platform, while in other cases, they act as notifications. All hooks are designed to be extended through `Open edX Django plugins`_ and configurations.
+
+Hooks can be of two types: events and filters. Events are signals sent in specific places whose receivers can extend functionality, while filters are functions that can modify the application's behavior. To allow extension developers to use these definitions in their implementations, both kinds of hooks are defined in lightweight external libraries:
+
+* `openedx-filters`_
+* `openedx-events`_
+
+The main goal of the framework is that developers can use it to change the platform's functionality as needed and still migrate to newer Open edX releases with little to no development effort. So, the framework is designed with stability in mind, meaning it is versioned and backward compatible as much as possible.
+
+A longer description of the framework and its history can be found in `OEP 50`_.
```

Comment:
```suggestion
A longer description of the framework and its history can be found in :doc:`openedx-proposals:architectural-decisions/oep-0050-hooks-extension-framework`
```

This will make it a bit more resilient of a link

Reply:
Would this work for all documents under docs.openedx.org? I'm not sure how this cross referencing works.

Reply:
Thank you for the suggestion! Now I'm working with the intersphinx mapping instead: https://github.com/openedx/docs.openedx.org/pull/599/commits/c953566eae941823a601d68f712856d501fd5119